tests/unit/test_charm.py

```diff
@@ -730,6 +748,7 @@ def test_config_changed_fails_apply_relation_data_not_ready(harness):
             return_value=Status.NOT_ALL_QUORUM,
         ),
         patch("managers.config.ConfigManager.config_changed", return_value=True),
+        patch("workload.ZKWorkload.get_version", return_value=""),
```

Comment:
**todo:** I think as we have to do this everywhere, let's add it to the conftest.
Then, we can in tests that actually need it, mark them to use not patch.

For example in ZooKeeper:

```python
# conftest.py
@pytest.fixture(autouse=True)
def patched_idle(mocker, request):
    if "nopatched_idle" in request.keywords:
        yield
    else:
        yield mocker.patch(
            "events.upgrade.ZKUpgradeEvents.idle", new_callable=PropertyMock, return_value=True
        )
```
```python
# test_upgrade.py
@pytest.mark.nopatched_idle
@pytest.mark.parametrize("upgrade_stack", ([], [0]))
def test_run_password_rotation_while_upgrading(harness, mocker, upgrade_stack):
    harness.charm.upgrade_events.upgrade_stack = upgrade_stack
...
```

Reply:
Done in 199067067b990cf6c425c0a909891e2af147b910, thanks for the suggestion